src/analyses/goto_check.cpp

```diff
@@ -1877,7 +1873,11 @@ void goto_check(
 
   Forall_goto_functions(it, goto_functions)
   {
-    goto_check.goto_check(it->second);
+    const symbolt *sym;
+    irep_idt mode;
+    if(!ns.lookup(it->first, sym))
```

Comment:
Shouldn't it be an invariant that the symbol exists? I think it should be safe to use
```
irep_idt mode=ns.lookup(it->first).mode;
```
?